### DISCUSSION TITLE
Welcome to SOLIDWORKS-Linux-Wine-Version- Discussions!

### DISCUSSION THREAD

**Discussion by cryinkfly** [cryinkfly/SOLIDWORKS-for-Linux#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by LikDev-256** [comment]
This is a good idea @cryinkfly.
I am currently extracting the portable I just thought to experiment 😅
The extraction is going to take a whole day I think 🙄

I'll try somehow to run it......

**Reply by cryinkfly** [reply]
I have visit this website and when I read something or look at the site, then I have my doubts. 🧐

![Bildschirmfoto_2021-09-24_08-54-08](https://user-images.githubusercontent.com/79079633/134631446-97348686-4d22-4dcd-bd7e-7206f3992051.png)

**Reply by LikDev-256** [reply]
I didn't care about it but it didn't work
I would be able to contact the portablehouse guy to help with this...... 🤔

**Reply by LikDev-256** [reply]
It did run in my dual booting windows 11 but stuck on loading registry I think there is a problem with the flexnet server

**Reply by LikDev-256** [reply]
https://t.me/thehouseofportable

**Reply by cryinkfly** [reply]
Okay. :+1:

**Reply by thehouseofportable** [reply]
Maybe not the best idea guys. It’s already a miracle that the portable works in Windows, since it’s a really huge and complex package. And sometimes it doesn’t even work there, let alone through Wine.

But anyway, the flexnet server is definitely your biggest problem imo, not sure if Wine is that good at virtualizing windows services. And SW uses that as licensing, very complex system behind the hood and no way around it.

**Comment by LikDev-256** [comment]
Yes @cryinkfly you are right This project is not about making illegal versions of the software so as you said using the portable version is a bad idea.

**Reply by LikDev-256** [reply]
I'll try a workaround. Without using cracked versions

**Reply by thehouseofportable** [reply]
Portable doesn’t mean cracked, it just means portable - no installation, ready to use 1-click, even from usb stick. Integrating the license into it is an optional step and is usually the hardest, so skipping it makes everything even easier.

**Comment by LikDev-256** [comment]
Thankfully you got my mail @thehouseofportable so can you help us with this project  ??!?

And sorry about that if I got something wrong and you must also update the agreement on the website it is actually sus as @cryinkfly doubted 😅

**Reply by thehouseofportable** [reply]
How is it sus? Maybe suggest how you’d update it.

And yes I can help but I don’t see the light out of the tunnel honestly.

**Reply by LikDev-256** [reply]
Ok never mind about what I said about your projects I apologize, ok. I think now, you're ok ?!?

**Reply by thehouseofportable** [reply]
Wutt? You said nothing wrong! 🤔 i was just asking suggestions as I don’t understand what you guys see wrong in it.

**Comment by LikDev-256** [comment]
Ok I'm getting directly to the problem.
The portable Solidworks is not working in even Windows it's just stuck in _loading registry_  I probably think that's a problem with the _Flexnet Server_ .

And I also wanna know how you tested it using lutris @cryinkfly 

Surely it is not running in wine there was some database problem. I think it is not supporting the .svm file format which has the data to run the executable.

**Reply by LikDev-256** [reply]
@thehouseofportable **reply please**

**Reply by thehouseofportable** [reply]
Ohhh yes I didn’t even think of that. Having an external svm file might also be a problem, not sure it does even read it. Depends on how wine works. But there’s no way to avoid that file since an executable can’t be bigger than 4gb...

SW works for me on a clean windows vm, if it gets stuck then it’s probably some flexnet-related junk in the host system, since many softwares use the same server for licensing. So if it’s already present then the procedure might not work, I tried once to clean everything i found in files and registry regarding things like flexnet/lmgrd and after cleaning everything it did work.

But again, everything under wine is different, might be a good idea to investigate/ask wine devs how wine manages windows services and external file access.

**Comment by LikDev-256** [comment]
**I tried running it but got a windows like this,**


![Screenshot_20210926_192908](https://user-images.githubusercontent.com/81100289/134812237-17432c5b-ee20-476d-b419-9394a92b697c.png)


**I think it probably means that the .exe is not supported
and a window like this, I think that the .exe data is inside the .svm file and wine is not supported to read .svm files ?!?? 🤔**


![Screenshot_20210926_200330-1](https://user-images.githubusercontent.com/81100289/134812316-7069e0e2-89fa-4a0e-9d7e-5522f315f41d.png)

**It didn't also load the thing I left it for about a hour 😅
As @thehouseofportable suggested this would mean that we must contact the wine devs to resolve this issue ?!??**

**Reply by LikDev-256** [reply]
And I think .svm is a libreoffice file extension and how did you stuff the program data in it ???!!? @thehouseofportable 

Why did you do it you could have just put folders

**Reply by thehouseofportable** [reply]
I didn’t decide it, I can’t decide folders or svm.
Svm is just the format created by the app I use to create portables which is turbo studio, and it’s the only format supported to have external data.
With turbo you can even open the svm files and see all the files and registry keys that are inside, svm is really just a raw pack of files, nothing fancy. So I’m also pretty sure that those ‘libreoffice svm’ files you mention are not the same as turbo’s svm, despite the same extension.

Interesting tho that the wine log doesn’t even load, it could help a lot understanding what’s going on :(

**Reply by LikDev-256** [reply]
Ya programs never like to debug itself and it will never reveal you it's secrets (log files) 🤣🤣

**Reply by LikDev-256** [reply]
Can you try to make the portable without using a .svm file ??
It would do a help .........

**Reply by thehouseofportable** [reply]
Sorry I already told you [here](https://github.com/cryinkfly/SOLIDWORKS-Linux-Wine-Version-/discussions/11#discussioncomment-1385843) that it’s not possible to avoid svm for apps larger than 4gb, since
 - data can’t be internal: .exe files cannot run if bigger than 4gb
 - so data must be external: there’s no other way to use external data, only svm I’m sorry, I didn’t create the portabilizing tools.

**Reply by LikDev-256** [reply]
@cryinkfly how did you run it in lutris ??!?

**Reply by LikDev-256** [reply]
Ok @thehouseofportable 
Then we must find a workaround 🤔

**Reply by cryinkfly** [reply]
Thank you very much for your time @thehouseofportable and that you give us more information about way to create a portable variant of Solidworks! :+1: 

But I disagree with you here:

> But again, everything under wine is different, might be a good idea to investigate/ask wine devs how wine manages windows services and external file access.

After all, I already got two similar programs to run on Linux:

- [Autodesk Fusion 360](https://github.com/cryinkfly/Fusion-360---Linux-Wine-Version-)
- [Rhino 3D](https://github.com/cryinkfly/Rhino-3D---Linux-Wine-Version-)

> data can’t be internal: .exe files cannot run if bigger than 4gb

> so data must be external: there’s no other way to use external data, only svm I’m sorry, I didn’t create the portabilizing tools.

The original Installer created by Solidworks use very simple way, because the program downloading all files, what we have selected, from the internet and when it’s done, then the installer install all these things step by step. 💡

And so we are not limited with this way! 🔓

@LikDev-256 Yes I have tested this also with Lutris but it doesn’t work. 😅

**Reply by thehouseofportable** [reply]
np I’m glad to help, I just think it’s hard since I know how tricky portables are, after making them for 11 years.

> After all, I already got two similar programs to run on Linux:
> * [Autodesk Fusion 360](https://github.com/cryinkfly/Fusion-360---Linux-Wine-Version-)
> * [Rhino 3D](https://github.com/cryinkfly/Rhino-3D---Linux-Wine-Version-)
> 

Every program is different from the other, even in the Adobe suite some work and some do not while being built in the exact same way, so who knows what’s going on under the hood. So, Rhino is pretty simple and expected to work, Fusion is also quite small and from what I remember it doesn’t use the same licensing as other autodesks since it’s mostly cloud-based in functionality they made it different. In both cases, they are smaller than 4gb so they didn’t require svm data file, which is one of the main problems with SW (as far as I think) together with flexnet server -which I think is a similar problem to the licensing you’d find if you try other autodesk’s using server licensing, like acad or revit.

> The original Installer created by Solidworks use very simple way, because the program downloading all files, what we have selected, from the internet and when it’s done, then the installer install all these things step by step. 💡
> And so we are not limited with this way! 🔓

I didn’t understand what you mean in all this part? :/
A portable can’t be made like an installer - i.e. (I think you mean) extracting an executable which then reads data from normal files/folders instead of a svm - because a portable is not simply all-files-packed-in-one, it also includes a virtualization part at execution which helps keeping the app as independent as possible from the host system, so for example when the app normally wants to write files to C: or the registry, when virtualized these attempts get all redirected to /sandbox/ folder to keep the edits local and the host system intact.

**Comment by LikDev-256** [comment]
I think there could be another issue here how does windows registry work in wine ?!?? @cryinkfly 
Pointed out by a post in quora,
_SW uses the Windows registry and so has to be just all ed on the target system, not just copied there_

**Comment by LikDev-256** [comment]
Hey is there a way to run a minimalized version of Windows like WSL ??!
Without using Virtualbox because running Windows 10 in it really suck.

And not resolving this issue is getting me off my robotics stuff because even though I have a dual bootable Windows system just opening the file explorer takes about 2 hour. I mean really.

@cryinkfly @thehouseofportable are we getting any further with this or we must switch to a new idea ??!?  🙄

**Reply by cryinkfly** [reply]
That would be worth a try, but what about the licensing? 👀

**Reply by cryinkfly** [reply]
### So we have [solved the problem](https://github.com/cryinkfly/SOLIDWORKS-Linux-Wine-Version-/issues/3) with installing VBA 7.1 (kb2783832)! ✅ 

But I have the next issue: https://github.com/cryinkfly/SOLIDWORKS-Linux-Wine-Version-/issues/12

**Reply by LikDev-256** [reply]
Ah.. @cryinkfly but It still doesn't seem to fix my issue though did you commit the changes ?!??!

**I was been in a hell for the past 2 days** because I idiotly thought it would be awesome to use the `Debian sid repo` and it broke my whole system and **I finally recovered from it now**. 
I think **no one had a problem like this except me** so I think that's a **problem with my distro ???!?** 
Or **_is it a good idea to just reinstall my system ?!?_** Because I was also having some **kernel panics** ....... recently 
I think I messed up something big.

To see if it goes away or a workaround is to install my distro in a VirtualBox and try to run it ??!

**Reply by Duedot43** [reply]
If you have been having kernel panics then that might be your problem have you had RAM issues? Is your hard drive going to sleep and not waking up? You might just want to back up your home folder and reinstall or do a full upgrade with sudo apt update sudo apt upgrade sudo apt dist-upgrade which I'm sure you already know of you use Kali right? VirtualBox has been giving me issues by uninstalling its kernel driver every time I reboot so I'll try to get that fixed so I can try installing it there.

**Reply by Duedot43** [reply]
so i got one of kali's vm images working and im trying to run the script but it will just not run so im just running the script section by section its installing wine at the moment ill see if it work's

**Reply by LikDev-256** [reply]
The kernel panic was caused because of the damn amdgpu why even these guys sell GPUs without drivers. I have reinstalled this system for like 20 times now. The only thing to make it work is reinstalling and even reinstalling is not a easy task like the bootable installer has a ton of bug you must literally make fixes in it when installing.

I actually use the use the GPU for CG stuff like my profile picture I made it from scratch using blender

But this is getting too out of topic so I'm like 75% on the side of reinstalling and I'll let know if it fixes the issue.

And I tried the flatpak installer it has a winetricks sha256 sum mismatch I fixed the sum in the `/usr/bin/winetricks` it seem to work with 
`winetricks -q vcrun2019`
But not with the `flatpak run `

**Reply by LikDev-256** [reply]
@Duedot43 what's the problem your having I could help you 🧐

**Reply by LikDev-256** [reply]
I just put the installer for download and it's gonna take an estimated 13hours and I'll reinstall in the morning I think .... 😶

**Reply by LikDev-256** [reply]
> If you have been having kernel panics then that might be your problem have you had RAM issues? Is your hard drive going to sleep and not waking up? You might just want to back up your home folder and reinstall or do a full upgrade with sudo apt update sudo apt upgrade sudo apt dist-upgrade which I'm sure you already know of you use Kali right? VirtualBox has been giving me issues by uninstalling its kernel driver every time I reboot so I'll try to get that fixed so I can try installing it there.

Can you explain Kernel driver ??!? Is it the VirtualBox drivers or other ??!

**Reply by Duedot43** [reply]
Just sometimes mostly after i reboot i need to re compile the virtualbox kernel drivers i got it fixed and kali running its just taking a while to upgrade kernel driver meaning how virtualbox emulates its VM's

**Reply by LikDev-256** [reply]
Are you using the VirtualBox file or the iso to install 
yeah upgrades take a lot of time

**Reply by Duedot43** [reply]
im using the virtualbox VDI from the kali website

**Reply by LikDev-256** [reply]
I haven't tested it before but if it's working no problem
**A tip to make the upgrade fast** just hold the security tools they aren't essential big ones like,
`sudo apt-mark hold burpsuite metasploit-framework maltego `

**Reply by Duedot43** [reply]
the update finished and i tried to install solidworks but it said solidworkssetup.exe is not supported on this system

**Reply by LikDev-256** [reply]
> the update finished and i tried to install solidworks but it said solidworkssetup.exe is not supported on this system

Whaaat ??!!
Why I used the script ?? Worked for me

**Comment by LikDev-256** [comment]
So... I reinstalled it for 2days fixing issues I eventually installed it and I setup the system how I liked and I restarted several times and the login screen is freezed and after changing sddm to gdm3 it kinda worked but it stuck after logging in I don't even have any clue like I checked every logs. And if someone can help me it would be great !!!

**Reply by cryinkfly** [reply]
@LikDev-256 Which graphics card and driver did you use for your system?

**Reply by LikDev-256** [reply]
@cryinkfly I use amdgpu-pro but no worries I just reinstalled another time what matters ... 😅

**Reply by LikDev-256** [reply]
Now I just want to do some setups and install wine and see

**Reply by cryinkfly** [reply]
Can you help me by this [issue](https://github.com/cryinkfly/SOLIDWORKS-for-Linux/issues/5#issuecomment-939268681)? (Specify the links to the other versions!)

**Comment by LikDev-256** [comment]
I tested it but now I can't even get past the starting installer the first thing is the close button is grayed out.....

**Comment by Duedot43** [comment]
I found a thing when solidworks "hangs" on something like the Microsoft sql server where there is no X button I've found if you open the windows task manager in the wineprefix and quit the install manager it comes up with a "would you like to quit?" If you say no it gets out of the install screen and shows you something looked like an install completed screen I'm reinstalling the wineprefix and trying again because i accidentally hit yes may be something to look into

**Reply by LikDev-256** [reply]
How do I install the WIndows task manager using wine

You mean that I have to run Solidworks using `WINEPREFIX=path/to/taskmanagerprefix wine ~/.wineprefixes//solidworks/setup.exe`

Or you tell that I have to run taskmanager and solidworks at the same time ?!?
Can you explain better ??!

**Reply by LikDev-256** [reply]
I was able to make a installation of windows in VirtualBox with the base system taking only 2GB ram and 11GB storage and 1core CPU it runs so much smoother but I still didn't install Solidworks I think these do not come to the minimal requirements of it.

**Reply by Duedot43** [reply]
Firstly SOLIDWORKS will not install into a VM it sees that's it's a vm and prevents the install and second what I meant by running the task manager is using the winetricks GUI and making a wineprefix there in the wineprefix menu there is an option to launch taskmgr.exe

**Reply by cryinkfly** [reply]
@LikDev-256 And it‘s so that you need a Network-License for Solidworks when you will install it into a VM!

And also you need a Windows-License. 😉

**Reply by LikDev-256** [reply]
I got the dual-boot working I used the win10debloat script to get some of them done
See it here I made some tweaks,
https://github.com/LikDev-256/win10script

I am also working on a script to easily install and update the latest versions of firefox
https://github.com/LikDev-256/Firefox-Installer_Firefox-Updater
Not still done yet.

Sooo I don't think I still need this but even though I'll try to help in getting rid of these issues. 😊

**Reply by cryinkfly** [reply]
@LikDev-256 

> @LikDev-256 The sad thing is I had never bought a licence in my life and I am not a highschool student nor our schools teach or provide SOLIDWORKS licence in our country a normal SOLIDWORKS licence cost about 6 months of salary my parents alone produce. I am only using pirated versions of them. In my country they don't check for software license for other than creations of patents or other commercial stuff so no worries. 😅😐

But you know that there is a [community](https://www.solidworks.com/support/community-download#no-back) and [private version](https://www.mbcad.de/solidworks-produkte/3dexperience-solidworks-privatlizenz/)? 💡

**The fact is: I will not support any pirated versions of them!** ⚠️

**Comment by LikDev-256** [comment]
@cryinkfly Thanks for the fact. I was able to obtain an SDK/SEK-ID from my school. I was able to get away with the thing that I am a member of the robotics club. The club is pretty much dead so they have closed the club for a long time. 🥳🥳🥳 

And another thing I am currently in Manjaro and it doesn't work here either I was testing this in Ubuntu also the same issue. I don't think the problem is distro-specific.
Still, no one had the same issue ?!?? Quit a weird...

Or it could be that I'm doing something wrong ?!?

**Comment by LikDev-256** [comment]
🥳🥳🎉

Thanks for the technique @Duedot43 It kinda worked till still the serial numbers but now I have a popup window. I think I'm still missing some packages because of that the popup is empty.

The screenshots,
![Screenshot_20211021_192848](https://user-images.githubusercontent.com/81100289/138293829-8864065e-e922-4912-bdc3-1146fe9cbfc5.png)
![Screenshot_20211021_192741](https://user-images.githubusercontent.com/81100289/138293913-dc9c41c6-9757-40ee-aab4-f3da466251b5.png)

**Reply by Duedot43** [reply]
Nice! With that empty dialog window just press enter as I said in a previous issue and it Will get you past.

**Reply by LikDev-256** [reply]
So we just close the issue I made ?!?

**Reply by Duedot43** [reply]
And to install the packages you need open the terminal and type this command 

WINEPREFIX=/path/to/prefix winetricks -q -f corefonts dotnet48 msxml6 dxvk win10

That should be all the packages you need as mentioned in the script and from one of my previous issues about solidworks not launching

**Reply by Duedot43** [reply]
Yeah I guess

**Reply by Duedot43** [reply]
Sorry I didn't get this solved earlier you did not mention that you had that empty dialog window that window should say some kind of XML error meaning it can't connect to solidworks and validate your license and that is solved by installing these packages.

**Reply by LikDev-256** [reply]
The empty dialog is new it only appeared after running it using the task manager

**Comment by Saleque474** [comment]
Good Job

**Comment by oldmanzwing** [comment]
Have a question, will this script work with 3DEXPERIENCE SOLIDWORKS for Makers installer? is it the same and only a different license or something different entirely. im talking about the local install not the cloud based apps.
